pagopa/io-dev-api-server
# [IOPID-313] FastLogin

## Short description
This PR adds the fast login logic to dev server

## List of changes proposed in this pull request
- src/features/fastLogin/fastLoginHandler.ts: handler for fast login request.
- src/features/fastLogin/nonceHandler.ts: handler for nonce request.
- src/middleware/fastLoginMiddleware.ts: middleware that returns 401 on every call if the sessionToken is not valid.
- src/middleware/lollipopMiddleware.ts: added error 403 to lollipop middleware. The error is triggered only if assertionRef is configured to have an expiration date.
- src/payloads/login.ts: removed hard coded info about sessionToken.
- src/persistence/appInfo.ts: added os information.
- src/persistence/lollipop.ts: added logic to handle assertionRef expiration.
- src/persistence/sessionInfo.ts: added info about session and its management.
- src/routers/__tests__/server.test.ts: updated test.
- src/routers/features/fastLogin/index.ts: added router for fast login and nonce.
- src/routers/features/featureFlagUtils.ts: updated.
- src/routers/public.ts:  added some clear function to logout, optimized code.
- src/server.ts: added FastLogin router and middleware.
- src/types/config.ts: added configurations for assertionRef and token session validity.
- src/utils/date.ts: added a util function.

## How to test
- Use dev server as always and verify that everything works as expected.
- in config.js, set an expiration time for assertionRef and test lollipop calls. It should work until it expires.
- in config.js, set an expiration time for assertionRef and sessionToken, enable the feature flag and make sure the app sends the right header value for fast login. It should follow the fast login spec.
- with every fast login feature configured, try to perform a standard login by setting the login header in the app. It should work as usual.
- all incorrect configurations (for example an app that requires a FastLogin but the ff or session duration time has not been set on the server), should - rightly - give an error
